functions/arrow-functions.md

```diff
@@ -0,0 +1,239 @@
+| `()`                                                | In one-line arrow functions, the parameter syntax stays the same.                                                                                                           |
+| `=>`                                                | In one-line arrow functions, we still need the arrow between the parameter list and the function body.                                                                      |
+| `'hey there.'`                                      | In one-line arrow functions, we can exclude the curly braces and the `return` keyword. However, this expression **_must_** be on the same line as the rest of the function. |
+| `;`                                                 | This statement now must conclude with a semicolon.                                                                                                                          |
+
+<!-- ### Exercise
+
+Convert this into an arrow function and minimize the code typed.
+
+```javascript
+const hello = function () {
+  return "hello world";
+};
+``` -->
+
+## Syntax with Parameters
+
+### With Multiple Parameters or Zero Parameters
+
+Like "normal" functions, arrow functions have their parameters listed in the parentheses.
+
+With zero parameters, we leave the parentheses empty:
+
+```javascript
+const sayHelloWorld = () => "Hello World!";
+```
+
+With multiple parameters, they are comma-separated.
+
+```javascript
+const add = (x, y) => x + y;
+```
+
+### With One Parameter
+
+This arrow function takes a single number and returns that number squared:
+
+```javascript
+const squareNum = (number) => {
+  return number * number;
+};
+console.log(squareNum(3)); // 9
+```
+
+When we have an arrow function that takes in _one and only one_ parameter, the parentheses around the single parameter are optional.
+
+Consider this refactoring:
+
+```javascript
+const squareNum = (number) => {
+  return number * number;
+};
+console.log(squareNum(3)); // 9
+```
+
+If we don't add parentheses when there are zero or more than one parameter, we'll get a syntax error!
+
+### !callout-info
+
+## Bonus Refactor: One Parameter and One Line
+
+We can refactor our `squareNum` arrow function, taking advantage of its one parameter and one-line return:
+
+<!-- prettier-ignore-start -->
+```js
+const squareNum = number => number * number;
+```
+<!-- prettier-ignore-end -->
+
+### !end-callout
+
+## One-Line Arrow Functions Returning Objects
+
+Consider this `makePerson` function, whose responsibility is to return an object.
+
+```javascript
+const makePerson = function (id, name) {
+  return {
+    id: id,
+    name: name,
+  };
+};
+```
+
+When we refactor it to a one-line arrow function, _because it specifically returns an object_, we need to wrap our object in parentheses.
```

Comment:
reword to explain _why_ the braces of an object literal pose a problem.
remove this version an replace with markdown in following comment
no suggestion because code
```suggestion
```

Reply:
To rewrite it as a one-line arrow function, we might try the following:

```javascript
const makePerson = (id, name) => { id: id, name: name };  // does NOT work!
```

But this will not work as expected!

To JavaScript, the opening brace `{` of the object is indistinguishable from the opening brace we would write at the start of a function body.

Fortunately, we can resolve this ambiguity by wrapping the object literal in parentheses!